pages/assets/[id].tsx

```diff
@@ -1,71 +1,167 @@
+  const [currentTab, setCurrentTab] = useState<typeof tabs[0]>()
+  const [assetData, setAssetData] = useState<any>()
+  const [tabData, setTabData] = useState<any>()
 
   const router = useRouter()
 
   useEffect(() => {
     if (router.query.id) {
-      setCurrentTab(window.location.hash)
+      if (!window.location.hash) {
+        router.replace("#overview")
+      }
+
+      if (!assetData) {
+        (async () => {
+          try {
+            const res = await HttpClient.get(`/api/assets/${router.query.id}`, {
+              headers: { authorization: `Bearer ${window.localStorage.getItem("access_token")}` },
+            })
+            setAssetData(res.body)
+          } catch (error) {
+            console.error(`[AssetDetailView] Failed while getting asset ${router.query.id}`, error)
+          }
+        })()
+      }
+
+      setCurrentTab(tabs.find((tab) => tab.hash === window.location.hash))
     }
-  }, [router])
+  }, [assetData, router])
 
   useEffect(() => {
     if (currentTab && router.query.id) {
-      const endpoint = mapHashToApiEndpoint(router.query.id as string, currentTab);
-
       (async () => {
         try {
-          const res = await HttpClient.get(endpoint!, {
+          const res = await HttpClient.get(currentTab.getDataSrc(router.query.id as string), {
             headers: { authorization: `Bearer ${window.localStorage.getItem("access_token")}` },
           })
-          console.log(res)
+          setTabData(res.body)
         } catch (error) {
           console.error("[AssetDetailView] Failed while getting asset", router.query.id)
         }
       })()
     }
   }, [currentTab, router])
 
+  const mapTabKeyToEnum = useCallback((tab: string) => ({
+    overview: AssetTabs.Overview,
+    responsibilities: AssetTabs.Responsibilities,
+  })[tab], [])
+
   return (
     <main>
+      <Head>
+        <title>{assetData?.name ?? "Data Marketplace"}</title>
+      </Head>
+
       <Container>
-        <aside>
-          <nav>
-            <ul>
-              <li>
-                <Link href="#overview">
-                  <a>
-                    Overview
-                  </a>
-                </Link>
-              </li>
-              <li>
-                <Link href="#responsibilities">
-                  <a>
-                    Responsibilities
-                  </a>
-                </Link>
-              </li>
-            </ul>
-          </nav>
-        </aside>
-
-        <Typography variant="h1_bold">Hello, World!</Typography>
+        <Header>
+          {assetData && (
+            <>
+              <Typography variant="h1_bold" as="h1">
+                {assetData.name}
+              </Typography>
+
+              <Button>
+                <Icon data={shopping_cart_add} />
+                Add to cart
+              </Button>
+            </>
+          )}
+        </Header>
+
+        <Divider />
+
+        <ContentContainer>
+          <TabNavContainer>
```

Comment:
I guess I have to ask... Did you consider to use the eds tabs? That would also give us the correct accessibility information that is needed for tabs.

Reply:
I guess we could do that. I only worry that the ui becomes very "vertical" if that makes sense. I could make a quick Figma sketch that uses the eds tabs

Reply:
It's quite a huge change I guess. But if we don't do that, we would at least have to deal with some a11y things ourselves. We could put a card to the backlog and investigate this further, as there is probably more components that should ideally utilise EDS? As a side note, we didn't use the EDS tabs in Envis as well due to a lot of style changes instead we used reach tabs. :)

Reply:
I've drawn up a quick [Figma sketch](https://www.figma.com/file/54wzCBdXkEbWsAGXOVWLd6/?node-id=128%3A1318) that I can link to in the ticket.

The design works in my opinion, but as you said, it's not a trivial change.